src/measure_theory/measure/finite_measure_weak_convergence.lean

```diff
@@ -0,0 +1,214 @@
+/-
+Copyright (c) 2021 Kalle Kytölä. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kalle Kytölä
+-/
+import measure_theory.measure.measure_space
+
+/-!
+# Weak convergence of (finite) measures
+
+This file will define the topology of weak convergence of finite measures and probability measures
+on topological spaces. The topology of weak convergence is the coarsest topology w.r.t. which
+for every bounded continuous `ℝ≥0`-valued function `f` on, the integration of `f` against the
+measure is continuous.
+
+TODOs:
+* Define the topologies (the current version only defines the types) via
+  `weak_dual ℝ≥0 (α →ᵇ ℝ≥0)`.
+* Prove that an equivalent definition of the topologies is obtained using bounded continuous
+  `ℝ`-valued functions instead.
+* Include the portmanteau theorem on characterizations of weak convergence of (Borel) probability
+  measures.
+
+## Main definitions
+
+The main definitions are the types `finite_measure α` and `probability_measure α`.
+
+TODO:
+* Define the topologies on the above types.
+
+## Main results
+
+None yet.
+
+TODO:
+* Portmanteau theorem.
+
+## Notations
+
+No new notation is introduced.
+
+## Implementation notes
+
+The topology of weak convergence of finite Borel measures will be defined using a mapping from
+`finite_measure α` to `weak_dual ℝ≥0 (α →ᵇ ℝ≥0)`, inheriting the topology from there.
+
+The current first implementation of `finite_measure α` and `probability_measure α` is directly
+as subtypes of `measure α`, and the coercion to a function is the composition `ennreal.to_nnreal`
+and the coercion to function of `measure α`. Another alternative would be to use a bijection
+with `vector_measure α ℝ≥0` as an intermediate step. This can be changed if
```

Comment:
sentence not quite finished?

Reply:
Thank you --- and sorry! I guess I was undecided about whether to have a sentence here at all, and was tired when writing, so committed changes with a "compromise" half sentence.

Will fix.